common/io.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_COMMON_IO_HPP
+#define MENDER_COMMON_IO_HPP
+
+#include <common/error.hpp>
+#include <common/expected.hpp>
+
+#include <memory>
+#include <system_error>
+#include <vector>
+
+namespace mender::common::io {
+
+using namespace std;
+
+using mender::common::error::StdError;
+using mender::common::expected::Expected;
+
+class Reader {
+public:
+	virtual ~Reader() {};
+
+	virtual Expected<size_t, StdError> Read(vector<uint8_t> &dst) = 0;
```

Comment:
I prefer naming `using ...` these types in the libraries/namespaces that work with them.

Reply:
👍

This got rid of an additional `using` I had added later, so definitely worth it.